### ISSUE TITLE
`echo = None` breaks indentation in Stata `python:`…`end` blocks

### ISSUE THREAD

**Issue by ideabucket** [hugetim/nbstata#14]
Here's a fun one.

Try this:

```stata
%%echo
python:
try:
    print("This works!")
except:
    pass
end
```

Then try this:

```stata
%%noecho
python:
try:
    print("This raises an IndentationError.")
except:
    pass
end
```

I assume that the behind-the-scenes conversion nbstata is doing to make the cells into `program`s is doing something to the indentation that Python doesn't like.

A solution might be a `%%python` magic, which dumps the cell to a temporary `.py` file and runs it using `pystata.stata.run("python script tempfile.py")`, so the python still runs in the inside-Stata python context instead of the kernel environment?

(My use case is trying to offload fetching and parsing some remote JSON to python before passing it back to Stata, since Stata's JSON-handling tools are so terrible.)

**Comment by hugetim** [comment]
Yes, of course! Thank you for finding and reporting. That is something I'm explicitly doing. I should be able to instead just preserve indentation in the code-standardizing I'm doing behind the scenes for "noecho" mode.

**Comment by hugetim** [comment]
I think this is fixed in v0.6.2, but let me know if you see it again.

p.s. I would like to add a %%python magic, but I was thinking it would ideally drop you out to the ipython shell in which `pystata` runs rather than run the script via Stata. Your way may be simpler to code, though, so we'll see...
